Feat/15669195 activity feed table

TomaszChmielPagepro opened the pull request:
## Description

This PR contains changes from https://github.com/JoinColony/colonyCDapp/pull/1336 as this hook was needed for table implementation. The status column is not sortable for now as `useActivityFeed` does not support this.

**New stuff** ✨

Added colony activity table without filters as this PR has gotten pretty big already.

![image](https://github.com/JoinColony/colonyCDapp/assets/38806520/eb3c9633-5e98-4b5f-8f56-1e40f3eb2cd8)
![image](https://github.com/JoinColony/colonyCDapp/assets/38806520/e92ccdae-dfcf-4f69-b832-7029249d4d6f)

melyndav reviewed (changes requested):
Hey @TomaszChmielPagepro! Amazing work on this and great to see it come to life.

A few changes to align it with the design.

1. When I select the date arrow, it hides (I'm guessing while the data is being pulled) and then comes back again only when I try to select again. Can it show the whole time to avoid user confusion? 

![Animation111](https://github.com/JoinColony/colonyCDapp/assets/82127904/b94c616a-49a6-4cbc-a88d-ecfd829c0aff)

2. The 'forced' pill should be replaced with the passed version? I understand this isn't aligned with the design. 

![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/75e8907e-9ba1-4bec-9e96-5734038a391f)

Link to Figma if you need the component - https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=16368-84713&mode=design&t=qE26uhUdRpPwHXpn-4

3. Background hover on each row is missing (you can find the correct example on the incoming funds table). 

4. When a table has less than 10 results, the padding allowed for pagination and number of pages should be removed and matched with the design in Figma to remove the empty space at the bottom of the table. 

<img width="1060" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/07618ac8-274d-40c3-90ea-3f4b2bbb53f0">

Link to Figma for reference: https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=8039-377316&mode=design&t=qE26uhUdRpPwHXpn-4

5. The title of each action/motion should match Figma:
![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/99720889-cbf0-4d3b-aebf-b6424141c7a5)

6. There's an issue when switching between actions:
![switching-between-actions](https://github.com/JoinColony/colonyCDapp/assets/82127904/72614dc7-4eae-4d14-82ef-703cfa443515)

7. The share button in the dropdown menu when selected, should use the same copy tooltip that's in the dashboard PR on the share icon and also in the dropdown menu - https://github.com/JoinColony/colonyCDapp/pull/1315

<img width="126" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/5c79c55e-a1c8-4cb1-a28d-a402d6e81b06">

8. The View on Ethscan button should use the logic from this component src/components/shared/TransactionLink/TransactionLink.tsx

9. The status pills are not being picked up correctly. I made a motion but it's showing forced. 

10. Active action is not appearing in the list:
![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/11edc010-03e0-411a-835b-06e3e5c15c27)

11. Time is showing as a negative value
![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/2e9da323-59b9-4a7a-bb77-688203e875be)

12. When there are more than 10 items, the pagination shows but the Page # is missing bottom left.
<img width="1052" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/82127904/3944a8ab-52dc-4960-859b-58a000a9a17c">

13. When the motion is in the finalizable state, there is no timer. Here you can see the timer showing incorrectly.
![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/0c34048a-c915-4001-846d-12d4bb25f154)

14. I noticed this error in the console:
![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/3e633307-9610-43d6-b743-f494e97ed3e8)

15. When I scroll, there is no padding at the bottom of the page. Globally, there should be 
16px in the content to match the bottom of the left nav container.
![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/8b540a3a-f4ee-4c4d-9725-3b6cc4d8ab5f)

16. I'm seeing double scrollbars on the right
![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/de42f4c6-17d6-4cd9-bc09-53da120beec5)

rdig reviewed (changes requested):
- Shouldn't this PR based on `feat/motion-status-filtering` and not `master`, seeing as it's based on #1336 ? Otherwise, merging this will be quite problematic

- Longer domain names break the UI
![Screenshot from 2023-11-23 13-04-36](https://github.com/JoinColony/colonyCDapp/assets/1193222/56d77f83-3fdd-4b25-b57d-440e6659b879)

- The action with `zzz` as a title is a Motion, the other one is a plain action. There should be some distinction between them
![Screenshot from 2023-11-23 13-04-53](https://github.com/JoinColony/colonyCDapp/assets/1193222/68da2ed8-a371-40f4-9bbc-bdcae7df48ff)

TomaszChmielPagepro commented:
@rdig 
1. This is not the case anymore as `feat/motion-status-filtering` has been merged to master. I have rebased my branch on master as well.
2. Elipsis added
3. @arrenv told me that you guys have discussed this internally and it's not an issue.

TomaszChmielPagepro commented:
Thanks for the review @melyndav 

1. I have disabled clearing sort state
2. Forced changed to Passed
3. Background added
4. Removed empty space
5. Font style updated
6. There is a related PR for action sidebar closing which may have an impact on this. I have added some cases for initial values loading in `src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts` but it needs more fixes as not every action is handled properly and I don't want to make this PR too big
7. I will fix this when the PR mentioned by you is merged because it seems like some logic may be shared here (or if it will take too long to merge this PR I will just do it here anyway tomorrow)
8. Used `TransactionLink` component
9. I couldn't reproduce that. Can you please share more details here?
10. I have added refetch on action creation but we may need to refetch the list on motion state changes as well. I will try to add this tomorrow.
11. Fixed
12. We don't have information about the total page count. Do you want me to add the current page here only?
13. When the motion state is `Finalizable` the timer will show negative time as there is no timer value for this state in the API. Am I missing something here?
![image](https://github.com/JoinColony/colonyCDapp/assets/38806520/ca460b81-d957-4ec4-9079-929a28aca52f)
14. Fixed
15. Padding added
16. Is there any interaction that could cause this? I can't reproduce this

melyndav commented:
Hey @TomaszChmielPagepro! To answer some of your questions - 

> 9. I couldn't reproduce that. Can you please share more details here?
Leave this for now and we'll see if we can pick this up with further testing. 

> 12. We don't have information about the total page count. Do you want me to add the current page here only?
Yes please. So it will display Page 1 for example.

> 13. When the motion state is `Finalizable` the timer will show negative time as there is no timer value for this state in the API. Am I missing something here?
Can we hide it when it's a negative value?

> 16. Is there any interaction that could cause this? I can't reproduce this
I couldn't replicate this again. I'll keep on eye it.

melyndav reviewed (changes requested):
Hey @TomaszChmielPagepro super work on the updates I requested, so thank you!

Just a few more - 

1. The pagination (page #) shouldn't show when there is 10 or less results. And the extra padding at the bottom should be removed like Figma: https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=8039-377316&mode=design&t=qE26uhUdRpPwHXpn-4

2. Row heights should be 58px to keep it consistent - 
![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/3d3f8ffd-8aa6-4299-9fee-c2805a5aaa4e)

3. For icons, please use the phosphor icon library - 

![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/54f0138b-ba6b-48e0-94f1-332d19bfadde)
 
 4. The share menu item in each row's dropdown is missing the tooltip copied state as mentioned previously:
 
_The share button in the dropdown menu when selected, should use the same copy tooltip that's in the dashboard PR on the share icon and also in the dropdown menu_ - https://github.com/JoinColony/colonyCDapp/pull/1315

![image](https://github.com/JoinColony/colonyCDapp/assets/82127904/08b67c61-edb4-4a29-ba58-94b1a4600fa2)

5. The View on Ethscan button should use the logic from this component (as mentioned in previous comment) 
src/components/shared/TransactionLink/TransactionLink.tsx

arrenv commented on src/components/v5/common/ActionSidebar/partials/forms/AdvancedPaymentForm/utils.tsx:
```diff
@@ -28,35 +29,30 @@ const getTokensText = (
-      return 'multiple tokens';
+      return 'with multiple tokens';
     case 1:
-      return '1 token';
+      return 'with 1 token';
     default:
-      return `${tokensCount} tokens`;
+      return `with ${tokensCount} tokens`;
```
We should be using translation here, right?

arrenv commented on src/components/v5/common/ActionSidebar/partials/forms/MintTokenForm/utils.tsx:
```diff
@@ -1,83 +1,32 @@
+        : undefined,
+    },
+    {
+      [ActionTitleMessageKeys.Amount]: '',
+      [ActionTitleMessageKeys.TokenSymbol]: 'native tokens',
```
Translation issue here as well.

arrenv commented on src/components/v5/common/ActionSidebar/partials/forms/SplitPaymentForm/utils.tsx:
```diff
@@ -1,88 +1,41 @@
+    },
+    {
+      [ActionTitleMessageKeys.Recipient]: '',
+      [ActionTitleMessageKeys.Amount]: 'an unspecified amount of',
+      [ActionTitleMessageKeys.TokenSymbol]: '',
```
Translation.

arrenv commented on src/utils/colonyActions.ts:
```diff
@@ -535,66 +536,36 @@ export const normalizeRolesForAction = (
-    roleTitle += normalizedRoles.length > 1 ? ' permissions' : ' permission';
+  if (isEmpty(assignedRoles)) {
+    return {
+      direction: 'Remove',
```
Translation.

arrenv commented on src/utils/colonyActions.ts:
```diff
@@ -535,66 +536,36 @@ export const normalizeRolesForAction = (
   return {
-    roleTitle,
-    direction,
+    direction: 'Assign',
```
Translation.

arrenv reviewed (changes requested):
Great work on this @TomaszChmielPagepro, I picked up a couple of missing translations.

TomaszChmielPagepro commented:
@melyndav I have fixed your comments

TomaszChmielPagepro commented:
@arrenv Translations added

arrenv commented on src/images/icons/arrow-down.svg:
This can be removed now, right?

TomaszChmielPagepro replied:
![image](https://github.com/JoinColony/colonyCDapp/assets/38806520/01e6f276-7876-4b13-858d-8431df442eea)
![image](https://github.com/JoinColony/colonyCDapp/assets/38806520/4a4e8d0d-d445-46da-b968-7c121f73f5b6)
 Seems like it's still used

arrenv reviewed (changes requested):
Thank you @TomaszChmielPagepro. These values should be formatted.

![image](https://github.com/JoinColony/colonyCDapp/assets/33682027/c313a2d7-48ed-4cef-99c5-629b2f0c4219)

melyndav reviewed (approved):
Approved from my side @TomaszChmielPagepro great work on this. Love the skeleton loader! Thanks for all your work :)

arrenv reviewed (approved):
Huge amount of work here, everything I tested showed it working as expected. Very nice work! Thank you @TomaszChmielPagepro.

ArmandoGraterol commented on src/components/common/ColonyActions/ActionDetailsPage/DefaultAction/DefaultActionContent.tsx:
```diff
@@ -15,6 +15,7 @@ import TimeRelative from '~shared/TimeRelative/TimeRelative';
 import ActionAnnotation from './ActionAnnotation';
 
 import styles from './DefaultActionContent.css';
+import { useNetworkInverseFee } from '~hooks';
```
This import should be grouped with the other relative imports

ArmandoGraterol commented on src/utils/colonyActions.ts:
```diff
@@ -17,6 +17,7 @@ import { ColonyActionRoles, ColonyActionType } from '~gql';
 import { formatText } from './intl';
 import { MotionVote } from './colonyMotions';
 import { parseSafeTransactionType } from './safes';
+import { getRole } from '~constants/permissions';
```
Same here

ArmandoGraterol commented on src/components/v5/common/ActionSidebar/hooks/useActionDescriptionMetadata.tsx:
```diff
@@ -22,6 +26,10 @@ import { enterRecoveryModeDescriptionMetadataGetter } from '../partials/forms/En
 import { manageColonyObjectivesDescriptionMetadataGetter } from '../partials/forms/ManageColonyObjectivesForm/utils';
+import { ACTION_TYPE_FIELD_NAME } from '../consts';
+import { DescriptionMetadataGetter } from '../types';
+import { ActionTitleMessageKeys } from '~common/ColonyActions/helpers/getActionTitleValues';
+import UserPopover from '~v5/shared/UserPopover';
```
Same here

ArmandoGraterol reviewed (dismissed):
Good work! This is a huge one.

TomaszChmielPagepro commented:
@ArmandoGraterol import order updated
@rdig Please review your requested changes

rdig reviewed (changes requested):
- Motions that don't have at least 10% support should not show up in the activity list

![Screenshot from 2023-11-26 04-12-04](https://github.com/JoinColony/colonyCDapp/assets/1193222/fe298f1c-28c0-449f-b80f-4fa060bdeba3)

![Screenshot from 2023-11-26 04-12-12](https://github.com/JoinColony/colonyCDapp/assets/1193222/00c913df-192a-41cc-8795-7193df383cfd)

- Please use the correct explorer name / link contextual the current chain. There's util method that generates them for you.
![Screenshot from 2023-11-26 04-13-16](https://github.com/JoinColony/colonyCDapp/assets/1193222/cba4b837-e16d-4418-b035-3a1bb3dc39cd)
![Screenshot from 2023-11-26 04-15-43](https://github.com/JoinColony/colonyCDapp/assets/1193222/0b04d28e-73ba-417f-a6a4-d62ad8ecbf25)

TomaszChmielPagepro commented:
@rdig 
1. I forgot to reenable filter for this, fixed
2. It's done like this. It depends on your env variables which network appears here (this is done the same way as it was in old UI).
![image](https://github.com/JoinColony/colonyCDapp/assets/38806520/7825e94b-df43-4c45-acf1-62e871a54574)
![image](https://github.com/JoinColony/colonyCDapp/assets/38806520/ab78184b-433b-4d9d-bcd5-de7ce0987b7b)

rdig reviewed (changes requested):
After the latest changes only 6 actions show, when in fact I have over 100 created at this point:

![Screenshot from 2023-11-27 15-23-37](https://github.com/JoinColony/colonyCDapp/assets/1193222/b3a2c1a6-6026-406d-9f12-7907025c4537)
![Screenshot from 2023-11-27 15-26-32](https://github.com/JoinColony/colonyCDapp/assets/1193222/0830ba9a-9d71-4c28-8de0-4da5b673dd0a)

- "View on Etherscan" still shows up, despite me being on `ganache`

![Screenshot from 2023-11-27 15-25-03](https://github.com/JoinColony/colonyCDapp/assets/1193222/d62ec33a-26bf-4293-9311-a5bd274ac395)
![Screenshot from 2023-11-27 15-25-08](https://github.com/JoinColony/colonyCDapp/assets/1193222/11f841c6-0046-428d-a978-f8f2fd11b526)
![Screenshot from 2023-11-27 15-25-14](https://github.com/JoinColony/colonyCDapp/assets/1193222/c6835ad8-38ec-4a92-9054-9c2efb442a53)

TomaszChmielPagepro commented:
1. I don't know why this happens, in this PR I am just consuming `useActivityFeed` hook and in the latest changes I have just brought back `showInActionsList` filter
2. Fixed

arrenv reviewed (approved):
All actions are showing up on my end and "Noexplorer" is showing as expected. Thank you @TomaszChmielPagepro
